diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..212c933
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,13 @@
+cmake_minimum_required(VERSION 3.1)
+project(stb)
+
+set(library_name ${PROJECT_NAME})
+
+set(STB_HDRS stb.h stb_connected_components.h stb_c_lexer.h stb_divide.h
+stb_dxt.h stb_easy_font.h stb_herringbone_wang_tile.h stb_image.h
+stb_image_resize.h stb_image_write.h stb_leakcheck.h stb_perlin.h
+stb_rect_pack.h stb_textedit.h stb_tilemap_editor.h stb_truetype.h
+stb_voxel_render.h stretchy_buffer.h)
+set(STB_SRCS stb_vorbis.c)
+
+add_library(${library_name} ${STB_HDRS} ${STB_SRCS})
\ No newline at end of file
