diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..651ad32
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.1)
+project(stb)
+
+set(STB_LIB_NAME stb)
+
+set(STB_HDRS stb.h stb_c_lexer.h stb_connected_components.h stb_divide.h stb_dxt.h 
+stb_easy_font.h stb_herringbone_wang_tile.h stb_image.h stb_image_resize.h 
+stb_image_write.h stb_leakcheck.h stb_perlin.h stb_rect_pack.h stb_textedit.h 
+stb_tilemap_editor.h stb_truetype.h stb_voxel_render.h stretchy_buffer.h)
+set(STB_SRCS stb_vorbis.c stb_all.c)
+add_library(${STB_LIB_NAME} ${STB_HDRS} ${STB_SRCS})
\ No newline at end of file
diff --git a/stb_all.c b/stb_all.c
new file mode 100644
index 0000000..e73c4c4
--- /dev/null
+++ b/stb_all.c
@@ -0,0 +1,18 @@
+#include "stb.h"
+#include "stb_c_lexer.h"
+#include "stb_connected_components.h"
+#include "stb_divide.h"
+#include "stb_dxt.h"
+#include "stb_easy_font.h"
+#include "stb_herringbone_wang_tile.h"
+#include "stb_image.h"
+#include "stb_image_resize.h"
+#include "stb_image_write.h"
+#include "stb_leakcheck.h"
+#include "stb_perlin.h"
+#include "stb_rect_pack.h"
+#include "stb_textedit.h"
+#include "stb_tilemap_editor.h"
+#include "stb_truetype.h"
+#include "stb_voxel_render.h"
+#include "stretchy_buffer.h"
\ No newline at end of file
